dev/com.ibm.ws.security.javaeesec/src/com/ibm/ws/security/javaeesec/identitystore/LdapIdentityStore.java

```diff
@@ -302,61 +270,41 @@ private String getUserDn(String callerName, String filter, SearchControls contro
         } else {
+            DirContext ctx = null;
+            try {
+                ctx = bind();
+            } catch (NamingException e) {
+                Tr.error(tc, "JAVAEESEC_ERROR_EXCEPTION_ON_BIND", new Object[] { this.idStoreDefinition.getBindDn(), e });
+                throw new IllegalStateException(e);
+            }
             try {
-                if (filter.isEmpty()) {
-                    Tr.error(tc, "An empty callerSearchFilter was provided. Please specify a valid callerSearchFilter in @LdapIdentityStoreDefinition");
-                    return null;
-                }
-                DirContext ctx = bind();
                 if (tc.isDebugEnabled()) {
                     Tr.debug(tc, "JNDI_CALL search", new Object[] { searchBase, filter, printControls(controls) });
                 }
                 NamingEnumeration<SearchResult> ne = ctx.search(new LdapName(searchBase), filter, controls);
                 if (ne.hasMoreElements()) {
                     userDn = ne.nextElement().getNameInNamespace();
                     if (ne.hasMoreElements()) {
-                        Tr.error(tc, "Multiple principals were found using search base: " + searchBase + " and filter: " + filter);
+                        Tr.warning(tc, "JAVAEESEC_WARNING_MULTI_CALLER_LDAP", new Object[] { callerName, filter, searchBase });
                         return null;
                     }
-                } else {
-                    Tr.error(tc, "No principal was found using search base: " + searchBase + " and filter: " + filter);
-                    return null;
                 }
-            } catch (NamingException ne) {
-                Tr.error(tc, "A NamingException occurred", ne);
+            } catch (NamingException e) {
+                Tr.error(tc, "JAVAEESEC_ERROR_EXCEPTION_ON_SEARCH", new Object[] { callerName, filter, searchBase, e });
+                throw new IllegalStateException(e);
             }
         }
         return userDn;
     }
 
-    /**
-     * Get a user-readable string representing the {@link SearchControls} object.
-     *
-     * @param controls The controls to get the string for.
-     * @return The string representation for the SearchControls object.
-     */
-    private String printControls(SearchControls controls) {
-        StringBuffer result = new StringBuffer();
-        result.append("[searchScope: ").append(controls.getSearchScope());
-        result.append(", timeLimit: ").append(controls.getTimeLimit());
-        result.append(", countLimit: ").append(controls.getCountLimit());
-        result.append(", returningObjFlag: ").append(controls.getReturningObjFlag());
-        result.append(", returningAttributes: ").append(controls.getReturningAttributes()[0]).append("]");
-        return result.toString();
-    }
-
     private Set<String> getGroups(DirContext context, String userDn) {
```

Comment:
Add Javadoc. This applies to the class in general.

Reply:
I added javadoc there and fixed a few places that had outdated parameters.